packages/x-data-grid/src/tests/rowSpanning.DataGrid.test.tsx

```diff
@@ -0,0 +1,251 @@
+      const rowsWithSpannedCells = Object.keys(apiRef.current.state.rowSpanning.spannedCells);
+      expect(rowsWithSpannedCells.length).to.equal(1);
+      const rowIndex = apiRef.current.getRowIndexRelativeToVisibleRows('4');
+      expect(rowIndex).to.equal(1);
+      const spanValue = apiRef.current.state.rowSpanning.spannedCells['4'];
+      expect(spanValue).to.deep.equal({ code: 3, totalPrice: 3 });
+      const spannedCell = getCell(rowIndex, 0);
+      expect(spannedCell).to.have.style('height', `${rowHeight * spanValue.code}px`);
+    });
+  });
+
+  describe('filtering', () => {
+    it('should work with filtering when initializing filter', function test() {
+      if (isJSDOM) {
+        this.skip();
+      }
+      render(
+        <TestDataGrid
+          initialState={{
+            filter: {
+              filterModel: {
+                items: [{ field: 'description', operator: 'contains', value: 'Upgrade' }],
+              },
+            },
+          }}
+        />,
+      );
+      const rowsWithSpannedCells = Object.keys(apiRef.current.state.rowSpanning.spannedCells);
+      expect(rowsWithSpannedCells.length).to.equal(1);
+      const rowIndex = apiRef.current.getRowIndexRelativeToVisibleRows('5');
+      expect(rowIndex).to.equal(0);
+      const spanValue = apiRef.current.state.rowSpanning.spannedCells['5'];
+      expect(spanValue).to.deep.equal({ code: 2, totalPrice: 2 });
+      const spannedCell = getCell(rowIndex, 0);
+      expect(spannedCell).to.have.style('height', `${rowHeight * spanValue.code}px`);
+    });
+
+    it('should work with filtering when controlling filter', function test() {
+      if (isJSDOM) {
+        this.skip();
+      }
+      render(
+        <TestDataGrid
+          filterModel={{
+            items: [{ field: 'description', operator: 'contains', value: 'Upgrade' }],
+          }}
+        />,
+      );
+      const rowsWithSpannedCells = Object.keys(apiRef.current.state.rowSpanning.spannedCells);
+      expect(rowsWithSpannedCells.length).to.equal(1);
+      const rowIndex = apiRef.current.getRowIndexRelativeToVisibleRows('5');
+      expect(rowIndex).to.equal(0);
+      const spanValue = apiRef.current.state.rowSpanning.spannedCells['5'];
+      expect(spanValue).to.deep.equal({ code: 2, totalPrice: 2 });
+      const spannedCell = getCell(rowIndex, 0);
+      expect(spannedCell).to.have.style('height', `${rowHeight * spanValue.code}px`);
+    });
+  });
+
+  describe('pagination', () => {
+    it('should only compute the row spanning state for current page', async function test() {
+      if (isJSDOM) {
+        this.skip();
+      }
+      render(
+        <TestDataGrid
+          pagination
+          initialState={{ pagination: { paginationModel: { pageSize: 4, page: 0 } } }}
+          pageSizeOptions={[4]}
+        />,
+      );
+      expect(Object.keys(apiRef.current.state.rowSpanning.spannedCells).length).to.equal(0);
+      apiRef.current.setPage(1);
```

Comment:
`await act()` wouldn't have been enough here? It looks wrong that we need to use `waitFor`.